visualisation/nzvm.py

```diff
@@ -113,30 +123,25 @@ def display_data_matplot(data, num_x, num_y, max_color):
         print("Error: delta extends beyond boundaries of given data file")
         quit()
 
-    try:
-        load_data(args.vm)
-    except FileNotFoundError:
-        print("Error: VM file does not exist")
-        quit()
+    length, array = load_data(args.vm, nx, ny, nz)
 
-    if len(array) != nx*ny*nz:
-        print("Error: Inconsistent VM with YAML file. Check model and yml and try again. Expected",nx*ny*nz,"values, got",len(array))
+    if length != nx*ny*nz:
+        print("Error: Inconsistent VM size. Check model and nx, ny, nz parameters and try again. Expected",nx*ny*nz,"values, got",len(array))
         quit()
-            
+
+    #yzx
     if axis == 'x':
-        for point in range(start_point[0], start_point[0] + delta):
-            for z in range(nz):
-                data.append(get_value(point, start_point[1], z) * 1000)
-        overall_da = overall_dx
+        data = array[start_point[1],:,start_point[0]:(start_point[0]+delta+1)] * 1000
     else:
-        for point in range(start_point[1], start_point[1] + delta):
-            for z in range(nz):
-                data.append(get_value(start_point[0], point, z) * 1000)
-        overall_da = overall_dy
-
-    display_data_matplot(data, int(len(data)/nz), nz, args.maximum)
+        data = array[start_point[1]:(start_point[1]+delta+1),:,start_point[0]] * 1000
```

Comment:
For the Multiply by 1000, either make the 1000 a constant. Like called. `Km to m conversion factor` or put a comment describing what the 1000 does / means.